include/Gaffer/MetadataAlgo.h

```diff
@@ -115,6 +115,22 @@ GAFFER_API bool getBookmarked( const Node *node );
 GAFFER_API bool bookmarkedAffectedByChange( const IECore::InternedString &changedKey );
 GAFFER_API void bookmarks( const Node *node, std::vector<NodePtr> &bookmarks );
 
+/// Numeric Bookmarks
+/// =================
+///
+/// Numeric bookmarks can be used to tag a node with a numeric value. Each value
+/// can be assigned to only a single node at any given time. Setting a numeric
+/// bookmark will consequently remove it from another node if the value has been
+/// in use already. Nodes can have only a single tag at a time.
+
+/// \undoable
+/// Throws if given bookmark is not in {1, ..., 9}. Pass a nullptr for removing the bookmark.
+GAFFER_API void setNumericBookmark( ScriptNode *script, int bookmark, Node *node );
+GAFFER_API Node *getNumericBookmark( ScriptNode *script, int bookmark );
+/// Returns 0 if the node isn't tagged, the bookmark's value otherwise.
+GAFFER_API int getNumericBookmark( const Node *node );
```

Comment:
This is the method that does the looping we talked about. I didn't know what to call it so it's an overloaded version of the other `getNumericBookmark` for now. In general, terminology is a little confusing here, isn't it? What exactly is a `numeric bookmark`? Is it the node you get back when providing the int? Is it the int? Is it the combination of the two? The method naming suggests that it's the node, but what do we call the int then? I've called it "tag" in the documentation, but calling that second method "getTag" or similar would be weird, too? Maybe you have a good idea.

Reply:
The way I see it, a bookmark is a property of the script, with the property being identified by the integer (which bookmark is it), and the value being the node. The fact that we store things differently behind the scenes doesn't matter, because the point of these utility methods is to abstract that away. So I would probably avoid the "tag a node" terminology because that's about the internal representation. Maybe something like this?

> Each script has a set of numeric bookmarks numbered 1-9, each of which can have a node assigned.